### PR TITLE
Update buttons

### PR DESCRIPTION
**Pull request by PhungNg**
## Description
[DG-676](https://payexjira.atlassian.net/secure/RapidBoard.jspa?rapidView=80&projectKey=DG&modal=detail&selectedIssue=DG-676)

Updated Buttons component.
Also updated background color for primary button on hover and focus

## How Has This Been Tested?
Ran all existing test, and updated snap

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have updated the **CHANGELOG** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -685,6 +695,78 @@ exports[`Documentation: Buttons Developer documentation renders 1`] = `
        attribute on these links (to prevent them from receiving keyboard focus) and use custom JavaScript to disable their functionality.
     </p>
   </Alert>
+  <h2>
+    Loading state
+  </h2>
+  <p>
+    Use the loading state to show users their action is processing by adding the .loading class to the
```
Comment (pthoang) [inline src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap:702]: `.loading` should be in a code tag?

File: src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap [hunk 2]
```diff
@@ -685,6 +695,78 @@ exports[`Documentation: Buttons Developer documentation renders 1`] = `
...
+  </h2>
+  <p>
+    Use the loading state to show users their action is processing by adding the .loading class to the 
+    <CodeTags
+      code="<button>"
+      type="primary"
+    />
+     element.
+  </p>
+  <ComponentPreview
```
Comment (pthoang) [inline src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap:709]: Some of the buttons have wrong color on the loading circle when hovering over with the mouse. (Applies to the disabled state)

File: src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -987,14 +1054,6 @@ exports[`Documentation: Buttons Overview renders 1`] = `
                     "icon": "cloud",
                   },
                 },
-                Object {
```
Comment (pthoang) [inline src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap:990]: Shouldn't loader be available even though loader documentation was added? Might be good to give developers easy access to all the options available

File: src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap [hunk 4]
```diff
@@ -966,7 +1033,7 @@ exports[`Documentation: Buttons Overview renders 1`] = `
           },
           Object {
             "component": <Button
-              label="Do something"
+              label="Type something"
```
Comment (pthoang) [inline src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap:1036]: I think "type something" is a bit awkward in this scenario (as it is a button)

File: src/less/components/button.less [hunk 5]
```diff
@@ -94,7 +94,7 @@
     // --------------------------------------------------
 
     &.btn-primary {
-        .button-variant(@btn-primary-color, @btn-primary-bg, @btn-primary-hover, @btn-primary-disabled-color, @btn-primary-disabled-bg, @brand-secondary);
+        .button-variant(@btn-primary-color, @btn-primary-bg, @yellow-faded, @btn-primary-disabled-color, @btn-primary-disabled-bg, @brand-secondary);
```
Comment (pthoang) [inline src/less/components/button.less:97]: Maybe change `@btn-primary-hover` to have same value as `@yellow-faded` instead

**Review by pthoang — CHANGES_REQUESTED** [review]
Nice! Just a few changes needed

File: src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap (hunk 4 above)
Comment (VikingTristan) [reply src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap:1036]: I second this. You're not typing something when you are clicking a button. Would it be possible to bring this up with the designers? Perhaps they have their own arguments.

Comment (PhungNg) [reply src/App/ComponentsDocumentation/components/Buttons/__snapshots__/index.test.js.snap:1036]: I totally agree. This is something I should have thought of myself. I will bring this up for the designers

**Pull request closed without merging**